core/core.go

```diff
@@ -153,6 +153,9 @@ func (c *Core) procAppendQueue() {
 
 	if c.procCounter > c_normalListProcCounter {
 		c.procCounter = 0
+		if len(c.appendQueue.Keys()) == 0 {
```

Comment:
This might not catch the case where the appendQueue has blocks, but missing a parent and cannot progress.

I think it should INFO print the queue size every ~1m.
I think it should DEBUG print the hashes & heights of each entry in the queue every ~1m